File: app/src/main/java/com/seafile/seadroid2/data/model/dirents/DirentRecursiveFileModel.java
package com.seafile.seadroid2.data.model.dirents;

public class DirentRecursiveFileModel {
    public String name;
    public String parent_dir;
    //net data id(file_id/folder_id)
    //when file is empty, id is 0000000000000000000000000000000000000000
    public String id;
    public String type;
    public long mtime;   // last modified timestamp
    public String permission;

    public long size;    // size of file, 0 if type is dir

    //lock
    public boolean is_locked;

    public boolean locked_by_me;
    public long lock_time;
    public String lock_owner;

    public String modifier_name;
    public String modifier_contact_email;

}

File: app/src/main/java/com/seafile/seadroid2/framework/model/dirents/DirentRecursiveFileModel.kt
package com.seafile.seadroid2.framework.model.dirents

import android.text.TextUtils
import com.seafile.seadroid2.framework.util.Utils

class DirentRecursiveFileModel {
    @JvmField
    var name: String? = null

    @JvmField
    var parent_dir: String? = null

    /**
     * Normalizes [parent_dir] to ensure it is non-null, non-empty and ends with "/".
     */
    fun getParent_dir(): String {
        var normalized = parent_dir
        if (normalized.isNullOrEmpty()) {
            normalized = "/"
        }

        if (!normalized.endsWith("/")) {
            normalized += "/"
        }

        parent_dir = normalized
        return normalized
    }

    /**
     * Net data id (file_id/folder_id). Empty file uses 40 zeros.
     */
    @JvmField
    var id: String? = null

    @JvmField
    var type: String? = null

    /**
     * Last modified timestamp (seconds).
     */
    @JvmField
    var mtime: Long = 0

    @JvmField
    var permission: String? = null

    /**
     * Size of file; 0 if directory.
     */
    @JvmField
    var size: Long = 0

    // lock status
    @JvmField
    var is_locked: Boolean = false

    @JvmField
    var locked_by_me: Boolean = false

    @JvmField
    var lock_time: Long = 0

    @JvmField
    var lock_owner: String? = null

    @JvmField
    var modifier_name: String? = null

    @JvmField
    var modifier_contact_email: String? = null

    fun isDir(): Boolean = TextUtils.equals(type, "dir")

    fun getFullFileName(): String? = Utils.pathJoin(parent_dir, name)
}
